app/models/history.rb

```diff
@@ -39,6 +39,15 @@ class History < ApplicationRecord
 
   belongs_to :patient
 
+  after_save :touch_patient
+  before_destroy :touch_patient
+
+  def touch_patient
+    # patient updated_at should not be updated if history was created by a system generated report reminder or contact attempt
+    patient.touch unless history_type == HISTORY_TYPES[:report_reminder] ||
```

Comment:
I think this should be decoupled a bit. Why not introduce a `is_user_driven?` method on this model or something (horrible name I know, but you get the idea). The method would tell you if a History instance was user driven or not.

You could probably even eliminate this whole method, and use a nice and succinct lambda on the callback above.

Reply:
I was thinking the exact same thing. I wrote something that tried to go back to adding an if condition to `:touch` itself which I don't think is possible so I didn't mention it. But if the whole thing is just an `after_create` then the logic can come in.

Reply:
I too went down that "can I use a lambda or a proc with touch?" and sadly was disappointed...

Reply:
yeah that's what I tried to do as well and wasn't able to find any way to do a conditional touch, are we still able to simplify this by introducing a `user_driven?` method?

Reply:
https://guides.rubyonrails.org/active_record_callbacks.html#using-if-and-unless-with-a-proc is what I was thinking

Reply:
This is what I was thinking: https://en.wikibooks.org/wiki/Ruby_on_Rails/ActiveRecord/Callbacks#Proc

Reply:
how's this? https://github.com/SaraAlert/SaraAlert/pull/683/commits/792a5ecd740a4f5aa5729400b0edd21b90940a9a

Reply:
Move the logic to its own method. That lambda should only be one line. Also really should figure out how to not need to ignore any rubocop complaints.

Sorry to be nit-picky, but this is the first time we've done something like this for this app - and it's important to establish a good pattern for others to follow moving forward.

Reply:
no worries, I'm just a bit unsure how exactly this should be factored out as its own method. wouldn't doing that essentially revert it back to how it was before (`touch_patient`)? or did you mean something like this?

```
  # patient updated_at should not be updated if history was created by a system generated action
  after_create proc { |history| history.patient.touch unless system_generated_action? }

  def system_generated_action?
    history_type == HISTORY_TYPES[:report_reminder] || history_type == HISTORY_TYPES[:monitoree_data_downloaded] ||
    (history_type == HISTORY_TYPES[:contact_attempt] && created_by == 'Sara Alert System')
  end
```

Reply:
https://github.com/SaraAlert/SaraAlert/pull/683/commits/f2b282b88dc4c87fefd179875eb1f7157ac72b18